From 240fed3bc488b5bb73466dde6e297eecb8e4b99b Mon Sep 17 00:00:00 2001
From: Jackie Huang <jackie.huang@windriver.com>
Date: Wed, 8 Feb 2023 01:19:56 -0500
Subject: [PATCH 05/12] helm: add dl_files_arm64 and fix the rules

The dl_files for arm64 are different whith the ones for x86,
so add dl_files_arm64 for arm64.

Use DEB_HOST_ARCH to replace the hardcoded arch name in the
rules file.

Test Plan:
PASS: build-pkgs on x86-64 host
PASS: build-image on x86-64 host
PASS: build-pkgs on arm64 host
PASS: build-image on arm64 host
PASS: Deploy AIO-SX on x86-64 target
PASS: Deploy AIO-SX on arm64 target
PASS: Deploy AIO-DX on arm64 targets
PASS: Deploy std (2+2+2) on arm64 targets

Story: 2010739
Task: 47981

Depends-On: https://review.opendev.org/c/starlingx/root/+/889686

Change-Id: Ie71073e590fdde8ecf8d10d08c31e07904d89964
Signed-off-by: Jackie Huang <jackie.huang@windriver.com>
---
 kubernetes/helm/debian/deb_folder/rules | 5 +++--
 kubernetes/helm/debian/meta_data.yaml   | 5 +++++
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/kubernetes/helm/debian/deb_folder/rules b/kubernetes/helm/debian/deb_folder/rules
index 74e9f7e0..4344b6c9 100755
--- a/kubernetes/helm/debian/deb_folder/rules
+++ b/kubernetes/helm/debian/deb_folder/rules
@@ -6,9 +6,10 @@ export ROOT = debian/tmp
 export SBINDIR = $(ROOT)/usr/sbin
 export SUDOERDIR = $(ROOT)/etc/sudoers.d
 
+export DEB_HOST_ARCH = $(shell dpkg-architecture -qDEB_HOST_ARCH 2>/dev/null)
 export HELM_VERSION = 3.14.4
-export HELM_EXECUTABLE = linux-amd64/helm
-export HELM_PKG = helm-v$(HELM_VERSION)-linux-amd64.tar.gz
+export HELM_EXECUTABLE = linux-$(DEB_HOST_ARCH)/helm
+export HELM_PKG = helm-v$(HELM_VERSION)-linux-$(DEB_HOST_ARCH).tar.gz
 
 %:
 	dh $@
diff --git a/kubernetes/helm/debian/meta_data.yaml b/kubernetes/helm/debian/meta_data.yaml
index e4f7687e..25b41109 100644
--- a/kubernetes/helm/debian/meta_data.yaml
+++ b/kubernetes/helm/debian/meta_data.yaml
@@ -7,6 +7,11 @@ dl_files:
     topdir: linux-amd64
     url: https://get.helm.sh/helm-v3.14.4-linux-amd64.tar.gz
     sha256sum: a5844ef2c38ef6ddf3b5a8f7d91e7e0e8ebc39a38bb3fc8013d629c1ef29c259
+dl_files_arm64:
+  helm-v3.14.4-linux-arm64.tar.gz:
+    topdir: linux-arm64
+    url: https://get.helm.sh/helm-v3.14.4-linux-arm64.tar.gz
+    sha256sum: 113ccc53b7c57c2aba0cd0aa560b5500841b18b5210d78641acfddc53dac8ab2 
 
 revision:
   dist: $STX_DIST
-- 
2.30.2

